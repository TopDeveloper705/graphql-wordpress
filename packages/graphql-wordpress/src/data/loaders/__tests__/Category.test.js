import getCategoryLoaders from 'data/loaders/Category';

const Category = getCategoryLoaders();

describe('Test Category loader', () => {
  test('Load a category', async () => {
    const cat = await Category.load(13);
    expect(cat.getID()).toMatchSnapshot();
    expect(cat).toMatchSnapshot();
  });

  test('Load a category by slug', async () => {
    const cat = await Category.loadBySlug('watch-this');
    expect(cat).toMatchSnapshot();
  });

  test('Load categories', async () => {
    const cat = await Category.loadMany([13, 17]);
    expect(cat).toMatchSnapshot();
  });
});
