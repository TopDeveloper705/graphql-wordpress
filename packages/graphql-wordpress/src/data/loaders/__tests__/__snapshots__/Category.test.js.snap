// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Category loader Load a category 1`] = `"Q2F0ZWdvcnk6MTM="`;

exports[`Test Category loader Load a category 2`] = `
Category {
  "id": 13,
}
`;

exports[`Test Category loader Load a category by slug 1`] = `
Category {
  "slug": "watch-this",
}
`;

exports[`Test Category loader Load categories 1`] = `
Array [
  Category {
    "id": 13,
  },
  Category {
    "id": 17,
  },
]
`;
